### DISCUSSION TITLE
Save and reset form dialog

### DISCUSSION THREAD

**Discussion by gfo2007** [serenity-is/Serenity#6642] [Help] [answered]
Hi,
I want to be able to save a record and reset the form dialog for the next record to be entered instead of closing it. This will make it easy for users to make multiple entries without having to close the form dialog. 

Could anyone help with a code snippet?

**Comment by dfaruque** [comment, marked as answer]
protected getToolbarButtons(): Serenity.ToolButton[] {
            let buttons = super.getToolbarButtons();

            buttons.push({
                    title: 'Save & New',
                    icon: 'fa fa-save',
                    cssClass: 'btn-save-and-new',
                    onClick: () => {
                        this.save(response => {
                            this.loadEntity({} as any);
                        });
                    }
                });

            return buttons;
        }

**Reply by gfo2007** [reply]
Thanks @dfaruque . It worked perfectly.

I asked a few questions at the discussions of your Serenity Extra App but didn't get any reply. Thank you for all the extras you've added and if you get time, do offer a response to my question there.

Thank you.
